apis/v1beta1/targetallocator_rbac.go

```diff
@@ -0,0 +1,75 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+
+	"github.com/open-telemetry/opentelemetry-operator/internal/rbac"
+)
+
+var (
+
+	// targetAllocatorCRPolicyRules are the policy rules required for the CR functionality.
+	targetAllocatorCRPolicyRules = []*rbacv1.PolicyRule{
+		{
+			APIGroups: []string{"monitoring.coreos.com"},
+			Resources: []string{"servicemonitors", "podmonitors"},
```

Comment:
I'm adding [here](https://github.com/open-telemetry/opentelemetry-operator/pull/3103/files)  the Probe/ScrapeConfig support. I believe you can add both resources on CRPolicyRules.